### ISSUE TITLE
Multi-task Learning - Two instances of the same task

### ISSUE THREAD

**Issue by lashmore** [nyu-mll/jiant#1334]
I'd like to do multi-task learning on two instances of "sst", sentiment analysis. The first task learns to separate sentences into two classes (Classes: A and B), and the second task learns to separate sentences into 2 other, distinct classes (Classes: C and D). This is performed on the same data, and labels are provided for each of the instances of sst.

Currently, jiant seems to assume that the "multi-tasks" that are sharing an encoder are of a different task type. Is this true, and if not, how can I test jiant against a 2 instances of a single NLP task, like "sst"? The data_download function for example puts the GLUE dataset in the sst folder, but does not create a new sst folder for the second instance.

If I create two sst folders (see screenshot below), named sst1 and sst2, this also breaks the code for downstream config and run driver scripts

![Screen Shot 2021-08-17 at 5 59 13 PM](https://user-images.githubusercontent.com/8303867/129806322-62f6edc1-59a1-4b2b-bc85-e9f9138c5205.png)

**Comment by lashmore** [comment]
In general, there is a confounding of the usage of the term "Tasks" in the jiant documentation with "Dataset". GLUE is a dataset that does X, but there are many other datasets that also do X. 

I'm looking for a "task" that has a sentence and a binary label associated with each sentence. I'm going to be using my own private dataset (cannot be shared) for learning the two binary classification tasks.

**Comment by zphang** [comment]
Could you past the code you are using to run this? I agree that there's definitely some blurring of boundaries between datasets and tasks, but it should be possible to duplicate a task, give them different task names, and `jiant` should treat them as two separate tasks. (You can also configure if you would want them to share the same head or not.)

**Comment by lashmore** [comment]
How does one duplicate a task? This is what I did:

     import jiant.scripts.download_data.runscript as rs
     EXPERIMENT_DIRECTORY = "./content"
     TASKS = ["sst"]
     print(rs.SUPPORTED_TASKS)
     rs.download_data(task_names=TASKS, output_base_path=f"{EXPERIMENT_DIRECTORY}/tasks")
     export_model.export_model(hf_pretrained_model_name_or_path="roberta-base",
                              output_base_path="./content/models/roberta-base")

     # Then I copied the sst_config.json and make a "sentiment_config.json". I also created a "sentiment" folder in /content/tasks/data and put all of my private dataset's train, val and test jsonl files there.

    # Then I tried to tokenize
    TASKS = ["sentiment", "sst"]
    # Tokenize and cache each task
    for task_name in TASKS:
        tokenize_and_cache.main(tokenize_and_cache.RunConfiguration(
            task_config_path=f"./content/tasks/configs/{task_name}_config.json",
            hf_pretrained_model_name_or_path="roberta-base",
            output_dir=f"./cache/{task_name}",
            phases=["train", "val"],
        ))

**Comment by lashmore** [comment]
Update:

I found this: 
https://github.com/nyu-mll/jiant-v1-legacy/blob/master/tutorials/adding_tasks.md

And skipping the @register_task(), I was able to copy SstTask class and create a SaTask class that didn't import GlueMixin and had 4 classes instead of 2. I modified in particular the following code, and I'm getting the following error:

    class Example(BaseExample):
         ....
         def tokenize(self, tokenizer):
            return TokenizedExample(
                     ...
                    label_id = SaTask.LABEL_TO_ID[self.label]
            )

     class SaTask(Task):
         ....
         LABELS = ["Class1", "Class2", "Class3", "Class4"]
         LABEL_TO_ID, ID_TO_LABEL = labels_to_bimap(LABELS)        # THIS ERRORS OUT 


Error:
```
Traceback (most recent call last):
  File "jiant_main.py", line 35, in <module>
    tokenize_and_cache.main(tokenize_and_cache.RunConfiguration(
  File "/home/jovyan/jiant_nlp/jiant/proj/main/tokenize_and_cache.py", line 165, in main
    chunk_and_save(
  File "/home/jovyan/jiant_nlp/jiant/proj/main/tokenize_and_cache.py", line 58, in chunk_and_save
    full_chunk_and_save(
  File "/home/jovyan/jiant_nlp/jiant/proj/main/tokenize_and_cache.py", line 80, in full_chunk_and_save
    dataset = preprocessing.convert_examples_to_dataset(
  File "/home/jovyan/jiant_nlp/jiant/proj/main/preprocessing.py", line 119, in convert_examples_to_dataset
    data_rows = tokenize_and_featurize(
  File "/home/jovyan/jiant_nlp/jiant/proj/main/preprocessing.py", line 181, in tokenize_and_featurize
    data_rows = [
  File "/home/jovyan/jiant_nlp/jiant/proj/main/preprocessing.py", line 182, in <listcomp>
    example.tokenize(tokenizer).featurize(tokenizer, feat_spec)
  File "/home/jovyan/jiant_nlp/jiant/tasks/lib/sentiment.py", line 28, in tokenize
    label_id=SaTask.LABEL_TO_ID[self.label],
KeyError: 1
```

**Comment by lashmore** [comment]
Update:

The issue was my labels in my .jsonl files were integers, not strings, so LABELS should have been this:

    LABELS = [0,1,2,3]


Which makes the labels_to_bimap() function useless, but would be useful if I had defined my class labels as strings, especially strings that weren't able to be casted to integers.

The only other file I needed to modify in order to tokenize was jiant/tasks/evaluate/core.py. Had to add SaTask in the get_evaluation_scheme_for_task() function.
